lib/activesupport/all/activesupport.rbi

```diff
@@ -812,3 +812,338 @@ class ActiveSupport::Duration
   sig { params(precision: T.nilable(Integer)).returns(String) }
   def iso8601(precision: nil); end
 end
+
+# Implements a hash where keys `:foo` and `"foo"` are considered
+# to be the same.
+#
+# ```ruby
+# rgb = ActiveSupport::HashWithIndifferentAccess.new
+#
+# rgb[:black] = '#000000'
+# rgb[:black]  # => '#000000'
+# rgb['black'] # => '#000000'
+#
+# rgb['white'] = '#FFFFFF'
+# rgb[:white]  # => '#FFFFFF'
+# rgb['white'] # => '#FFFFFF'
+# ```
+#
+# Internally symbols are mapped to strings when used as keys in the entire
+# writing interface (calling `[]=`, `merge`, etc). This
+# mapping belongs to the public interface. For example, given:
+#
+# ```ruby
+# hash = ActiveSupport::HashWithIndifferentAccess.new(a: 1)
+# ```
+#
+# You are guaranteed that the key is returned as a string:
+#
+# ```ruby
+# hash.keys # => ["a"]
+# ```
+#
+# Technically other types of keys are accepted:
+#
+# ```ruby
+# hash = ActiveSupport::HashWithIndifferentAccess.new(a: 1)
+# hash[0] = 0
+# hash # => {"a"=>1, 0=>0}
+# ```
+#
+# but this class is intended for use cases where strings or symbols are the
+# expected keys and it is convenient to understand both as the same. For
+# example the `params` hash in Ruby on Rails.
+#
+# Note that core extensions define `Hash#with_indifferent_access`:
+#
+# ```ruby
+# rgb = { black: '#000000', white: '#FFFFFF' }.with_indifferent_access
+# ```
+#
+# which may be handy.
+#
+# To access this class outside of Rails, require the core extension with:
+#
+# ```ruby
+# require "active_support/core_ext/hash/indifferent_access"
+# ```
+#
+# which will, in turn, require this file.
+class ActiveSupport::HashWithIndifferentAccess < Hash
+  include Enumerable
+
+  extend T::Generic
+  K = type_member
+  V = type_member
+  Elem = type_member
+
+  sig { params(key: T.any(String, Symbol)).returns(T.untyped) }
```

Comment:
suggestion: make an internal type alias for `T.any(String, Symbol)` so you don't accidentally only allow one or the other on any method